db/db_impl/db_impl_open.cc

```diff
@@ -291,6 +292,29 @@ Status DBImpl::ValidateOptions(const DBOptions& db_options) {
         "writes in direct IO require writable_file_max_buffer_size > 0");
   }
 
+  if (db_options.daily_offpeak_start_time_utc != "" ||
+      db_options.daily_offpeak_end_time_utc != "") {
+    if (db_options.daily_offpeak_start_time_utc == "" ||
+        db_options.daily_offpeak_end_time_utc == "") {
+      return Status::InvalidArgument(
```

Comment:
I like the idea of putting them into a single option, too. Let me make the change.

Reply:
I agree that a single option is probably better.  Would it also then be possible to define/document a format that allows multiple offpeak times?  For example, from 1AM-5AM and 7PM-11PM.  How this "vector" is separated is left up to you to define.

Reply:
Thanks for your comment. Initially, we considered the possibility of having multiple peak and off-peak periods throughout the day. However, after an internal discussion, we decided to opt for a single off-peak time period for the following reasons:

1. Managing too many peak and off-peak periods could introduce unnecessary complexity. If such a feature needs to be supported in the future, it might be more practical to provide an API like `SetPeak(bool peak)` for users to explicitly designate peak times at the start or end of those periods.

2. The primary objective of defining this time period is to enhance resource utilization by preprocessing low-priority tasks, such as TTL-based compactions, during off-peak hours. This approach minimizes the chances of these tasks competing for resources with read and write operations during peak hours. Therefore, establishing a single off-peak time frame aligns better with this goal compared to having multiple peak times throughout the day. This approach is similar to what HBase does (http://bigdatainourpalm.blogspot.com/2017/09/how-to-control-time-during-which-hbase.html).